pydantic/types.py

```diff
@@ -516,3 +528,37 @@ def validate(cls, value: Union[str, bytes, int]) -> Union[IPv4Address, IPv6Addre
 
         with change_exception(errors.IPvAnyAddressError, ValueError):
             return IPv6Address(value)
+
+
+class IPvAnyInterface(_BaseAddress):
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: Union[str, bytes, int, NetworkType]) -> Union[IPv4Interface, IPv6Interface]:
+        try:
+            return IPv4Interface(value)
+        except ValueError:
+            pass
+
+        with change_exception(errors.IPvAnyInterfaceError, ValueError):
+            return IPv6Interface(value)
+
+
+class IPvAnyNetwork(_BaseNetwork):  # type: ignore
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: Union[str, bytes, int, NetworkType]) -> Union[IPv4Network, IPv6Network]:
```

Comment:
```suggestion
    def validate(cls, value: NetworkType) -> Union[IPv4Network, IPv6Network]:
```

Reply:
Ooops, I'm sorry for all that stuff that popped up again! It seems that something went wrong with my rebase. I'll fix that